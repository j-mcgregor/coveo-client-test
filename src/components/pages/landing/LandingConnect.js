import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import {
  fetchResults,
  selectProduct
} from '../../../flux/actions/queryActions';
import Main from './LandingContainer';

const mapStateToProps = state => ({
  results: state.results,
  errors: state.errors
});

const mapDispatchToProps = dispatch => ({
  fetchResults: query => dispatch(fetchResults(query)),
  selectProduct: product => dispatch(selectProduct(product))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Main));
