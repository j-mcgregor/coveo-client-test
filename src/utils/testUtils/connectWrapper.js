import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

export default (initialState, component) => {
  const mockStore = configureStore([thunk]);
  const store = mockStore(initialState);
  const wrapper = mount(
    <Provider store={store}>
      <BrowserRouter>{component}</BrowserRouter>
    </Provider>
  );

  return wrapper;
};
