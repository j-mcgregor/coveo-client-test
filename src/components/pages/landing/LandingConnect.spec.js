import React from 'react';
import connectWrapper from '../../../utils/testUtils/connectWrapper';
import LandingConnect from './LandingConnect';

describe('LandingConnect', () => {
  it('should render with the store', () => {
    const initialState = {
      results: {
        loading: false,
        errors: {},
        results: []
      },
      theme: {
        theme: 'light'
      }
    };
    const wrapper = connectWrapper(initialState, <LandingConnect />);

    expect(wrapper).toMatchSnapshot();
  });
});
