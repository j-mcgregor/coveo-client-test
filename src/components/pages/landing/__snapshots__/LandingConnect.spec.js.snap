// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingConnect should render with the store 1`] = `ReactWrapper {}`;
